crates/blockchain/payload.rs

```diff
@@ -581,87 +478,28 @@ fn apply_plain_transaction(
-    match EVM_BACKEND.get() {
-        Some(EVM::LEVM) => {
-            let store_wrapper = Arc::new(StoreWrapper {
-                store: context.evm_state.database().unwrap().clone(),
-                block_hash: context.payload.header.parent_hash,
-            });
-
-            let fork = context
-                .chain_config()?
-                .fork(context.payload.header.timestamp);
-            let blob_schedule = context
-                .chain_config()?
-                .get_fork_blob_schedule(context.payload.header.timestamp)
-                .unwrap_or(EVMConfig::canonical_values(fork));
-            let config = EVMConfig::new(fork, blob_schedule);
-
-            let report = backends::levm::execute_tx_levm(
-                &head.tx,
-                &context.payload.header,
-                store_wrapper.clone(),
-                context.block_cache.clone(),
-                config,
-            )
-            .map_err(|e| EvmError::Transaction(format!("Invalid Transaction: {e:?}")))?;
-            context.remaining_gas = context.remaining_gas.saturating_sub(report.gas_used);
-            context.block_value += U256::from(report.gas_used) * head.tip;
-
-            let mut new_state = report.new_state.clone();
-
-            // Now original_value is going to be the same as the current_value, for the next transaction.
-            // It should have only one value but it is convenient to keep on using our CacheDB structure
-            for account in new_state.values_mut() {
-                for storage_slot in account.storage.values_mut() {
-                    storage_slot.original_value = storage_slot.current_value;
-                }
-            }
-
-            context.block_cache.extend(new_state);
-
-            let receipt = Receipt::new(
-                head.tx.tx_type(),
-                report.is_success(),
-                context.payload.header.gas_limit - context.remaining_gas,
-                report.logs.clone(),
-            );
-            Ok(receipt)
-        }
-        // This means we are using REVM as default for tests
-        Some(EVM::REVM) | None => {
-            let report = backends::revm::execute_tx(
-                &head.tx,
-                &context.payload.header,
-                context.evm_state,
-                spec_id(
-                    &context.chain_config().map_err(ChainError::from)?,
-                    context.payload.header.timestamp,
-                ),
-            )?;
-            context.remaining_gas = context.remaining_gas.saturating_sub(report.gas_used());
-            context.block_value += U256::from(report.gas_used()) * head.tip;
-            let receipt = Receipt::new(
-                head.tx.tx_type(),
-                report.is_success(),
-                context.payload.header.gas_limit - context.remaining_gas,
-                report.logs(),
-            );
-            Ok(receipt)
-        }
-    }
+    let chain_config = context.chain_config()?;
```

Comment:
Not to solve now, but why could `chain_config` fail? It looks like something that should be checked at construction and always succeed afterwards. Am I missing something?